File: ifif/python/G5_20164.py
import sys

sys.stdin = open("input.txt","r")
N = input()
answer = [1e9, 0]


def solution(N, cnt):
    if len(N)==1:
        answer[0] = min(cnt, answer[0])
        answer[1] = max(cnt, answer[1])
        return
    if len(N)==2:
        N = str(int(N[0])+int(N[1]))
        return solution(N, cnt+checkOdd(N))
    for i in range(1, len(N)-1):
        for j in range(i+1, len(N)):
                a = int(N[:i])
                b = int(N[i:j])
                c = int(N[j:])
                num = checkOdd(a+b+c)
                solution(str(a+b+c), cnt+num)

def checkOdd(num):
    cnt = 0
    for n in str(num):
        if n in "13579":
            cnt+=1
    return cnt

solution(N, 0)
origin = checkOdd(N)
answer[0] += origin
answer[1] += origin
print(*answer)

File: ifif/java/G5_20164.java
// 홀수홀릭호석
// 구현

import java.io.*;

public class G5_20164 {
    static int minCnt = (int) 1e9;
    static int maxCnt = 0;

    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("input.txt"));
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        String num = sc.readLine();
        solution(num, 0);
        System.out.print(minCnt+countOdd(num));
        System.out.print(" ");
        System.out.println(maxCnt+countOdd(num));
    }

    private static void solution(String num, int cnt) {
        if (num.length() == 1) {
            minCnt = Math.min(cnt, minCnt);
            maxCnt = Math.max(cnt, maxCnt);
            return;
        } else if (num.length() == 2) {
            int intNum = Character.getNumericValue(num.charAt(0)) + Character.getNumericValue(num.charAt(1));
            num = String.valueOf(intNum);
            cnt += countOdd(num);
            solution(num, cnt);
        }
        else {
            for (int i=1; i<num.length()-1; i++) {
                for (int j=i+1; j<num.length(); j++) {
                    String a = num.substring(0, i);
                    String b = num.substring(i, j);
                    String c = num.substring(j);
                    int intAbc = Integer.parseInt(a) + Integer.parseInt(b) + Integer.parseInt(c);
                    int tmp = countOdd(String.valueOf(intAbc));
                    solution(String.valueOf(intAbc), cnt+tmp);
                }
            }
        }
    }

    public static int countOdd(String number) {
        int cnt = 0;
        int n = Integer.parseInt(number);
        while (n>=1) {
            if ((n%10)%2 == 1) {
                cnt += 1;
            }
            n /= 10;
        }
        return cnt;
    }
}
